lang/LangSource/SC_CLIOptions.cpp

```diff
@@ -0,0 +1,288 @@
+#include "boost/program_options.hpp"
+#include <iostream>
+#include <string>
+#include "SC_TerminalClient.h"
+#include "SC_CLIOptions.hpp"
+
+#include <SC_Filesystem.hpp>
+#include <SC_LanguageConfig.hpp>
+#include <SC_Version.hpp>
+
+namespace po = boost::program_options;
+using namespace std;
+
+namespace SC_CLI {
+
+/// parses the CLI - if the program should terminate after parsing it will return true
+/// an exit code is passed by reference
+bool CLIOptions::parse(int argc, char* argv[], SC_TerminalClient::Options& terminalOptions, int& exitCode) {
+    exitCode = 0;
+    po::options_description description("Interpreter for sclang (SuperCollider language)");
+
+    auto genericDescription = buildGenericDescription();
+    auto terminalDescription = buildTerminalDescription(terminalOptions);
+    auto languageConfigDescription = buildLanguageConfigDescription();
+
+    // Hidden options, will be allowed both on command line and
+    // in config file, but will not be shown to the user.
+    po::options_description hiddenDescription("Hidden options");
+    // clang-format off
+    hiddenDescription.add_options()
+        ("input-file", po::value<string>(&mInputFile), "input file")
+        ("sc-args", po::value<vector<string>>(&mScArgs), "sc args")
+    ;
+    // clang-format on
+
+    // positional options are declared here but captured via hidden description
+    po::positional_options_description positionalOptions;
+    positionalOptions.add("input-file", 1);
+    positionalOptions.add("sc-args", -1);
+
+    po::options_description cmdLineOptions;
+    cmdLineOptions.add(genericDescription)
+        .add(terminalDescription)
+        .add(hiddenDescription)
+        .add(languageConfigDescription);
+
+    po::variables_map vm;
+    try {
+        store(po::command_line_parser(argc, argv).options(cmdLineOptions).positional(positionalOptions).run(), vm);
+        notify(vm);
+    } catch (po::unknown_option& e) {
+        cerr << e.what() << endl;
+        exitCode = 1;
+        return true;
+    } catch (po::invalid_option_value& e) {
+        cerr << e.what() << endl;
+        exitCode = 1;
+        return true;
+    }
+    if (mParsingFailed) {
+        exitCode = 1;
+        return true;
+    }
+
+    if (vm.count("sc-args")) {
+        terminalOptions.mArgs = vm["sc-args"].as<vector<string>>();
+    }
+
+    if (mPrintVersion) {
+        cout << "sclang " << SC_VersionString() << " (" << SC_BuildString() << ")" << endl;
+        return true;
+    }
+
+    // this is handled here and not in the description declaration b/c we need access to the declarations
+    if (mPrintHelp) {
+        // construct a options_description w/o hidden positional options
+        po::options_description visibleDescription("");
```

Comment:
Since you said you are ok with nitpicks, here's another one: no need to explicitly pass an empty string, just use the default constructor, as you do with the other instances of `po::options_description`.